source/MaterialXGraphEditor/Graph.cpp

```diff
@@ -3566,97 +3566,133 @@ void Graph::addNodePopup(bool cursor)
         // Input string length
         // Filter extra nodes - includes inputs, outputs, groups, and node graphs
         const std::string NODEGRAPH_ENTRY = "Node Graph";
-        for (std::unordered_map<std::string, std::vector<std::vector<std::string>>>::iterator it = _extraNodes.begin(); it != _extraNodes.end(); ++it)
+
+        std::vector<std::string> ordered_keys = {
```

Comment:
1. is this a reasonable way to approach this? is this annoying to update as new groups are added? how often are groups added?

2. I realize this probably isn't the best place for this list to live, is there a recommended place to keep a list like this?

Reply:
@beersandrew This choice seems fine to me, and further in the future we can move this string vector into a global constant that's accessible through the MaterialX API.  Often new MaterialX features are developed in this way, where initially they're local data or functions in an example application, and they're moved into the core API as a second step.

Reply:
Awesome, thanks for the note, I did move it up to a higher function per @bernardkwok's suggestion on Slack, still might not be the end location but I think it's better than where I had it in the first place